virtualhost.sh

```diff
@@ -68,24 +77,64 @@ if [ "$action" == 'create' ]
 		fi
 
 		### create virtual host rules file
-		if ! echo "
-		<VirtualHost *:80>
-			ServerAdmin $email
-			ServerName $domain
-			ServerAlias $domain
-			DocumentRoot $rootDir
-			<Directory />
-				AllowOverride All
-			</Directory>
-			<Directory $rootDir>
-				Options Indexes FollowSymLinks MultiViews
-				AllowOverride all
-				Require all granted
-			</Directory>
-			ErrorLog /var/log/apache2/$domain-error.log
-			LogLevel error
-			CustomLog /var/log/apache2/$domain-access.log combined
-		</VirtualHost>" > $sitesAvailabledomain
+		if ! echo '
+<VirtualHost '$ipAddress':80>
+	ServerAdmin '$email'
+    ServerName '$domain'
+    ServerAlias '$domain'
+    DocumentRoot '$rootDir'
+	<Directory />
+		AllowOverride All
+	</Directory>
+	<Directory '$rootDir'>
+		Options Indexes FollowSymLinks MultiViews
+		AllowOverride all
+		Require all granted
+	</Directory>
+	ErrorLog /var/log/apache2/'$domain'-error.log
+	LogLevel error
+	CustomLog /var/log/apache2/'$domain'-access.log combined
+</VirtualHost>
+<IfModule mod_ssl.c>
+    <VirtualHost _default_:443>
+    
+        ServerAdmin '$email'
+        ServerName '$domain'
+        
+        DocumentRoot '$rootDir'
+        
+        ErrorLog ${APACHE_LOG_DIR}/'$domain'-ssl-error.log
+        CustomLog ${APACHE_LOG_DIR}/'$domain'-ssl-access.log combined
+        
+        SSLEngine on
+        SSLCertificateFile /etc/apache2/ssl/'$domain'.crt
+        SSLCertificateKeyFile /etc/apache2/ssl/'$domain'.key
+
+        <Directory />
+			AllowOverride All
+		</Directory>
+		<Directory '$rootDir'>
+			Options Indexes FollowSymLinks MultiViews
+			AllowOverride all
+			Require all granted
+		</Directory>
+
+        <FilesMatch "\.(cgi|shtml|phtml|php)$">
+                        SSLOptions +StdEnvVars
+        </FilesMatch>
+
+        <Directory /usr/lib/cgi-bin>
+                        SSLOptions +StdEnvVars
+        </Directory>
+
+        BrowserMatch "MSIE [2-6]" \
+                        nokeepalive ssl-unclean-shutdown \
+                        downgrade-1.0 force-response-1.0
+        BrowserMatch "MSIE [17-9]" ssl-unclean-shutdown
+
+    </VirtualHost>
+</IfModule>
+' > $sitesAvailabledomain
```

Comment:
this is strange, since use SSL needs to use another port and ssl certificate path.